### ISSUE TITLE
[JENKINS-13972] Concurrent matrix builds abort

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/matrix-project-plugin#505]
<p>Reproduction case:</p>

<p>Create a concurrent matrix job with a user defined axis that does a 'sleep 120' as its build step. Launch several of these jobs, enough that all available executors are taken up and there are still builds in the queue. Some of these builds will abort, with a console message similar to:</p>

<p><span class="error">&#91;...&#93;</span><br/>
9 completed with result SUCCESS<br/>
24 completed with result SUCCESS<br/>
23 completed with result SUCCESS<br/>
2 completed with result SUCCESS<br/>
10 appears to be cancelled<br/>
10 completed with result ABORTED<br/>
25 appears to be cancelled<br/>
25 completed with result ABORTED<br/>
18 appears to be cancelled<br/>
18 completed with result ABORTED<br/>
13 appears to be cancelled<br/>
<span class="error">&#91;...&#93;</span></p>

<p>For my test case, I have 26 slaves, 82 executors, 25 sub-jobs. I can reproduce reliably if I launch 5 or more top level jobs at once.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jkoleszar">jkoleszar</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972" target="_blank">Concurrent matrix builds abort</a></i></summary>
<i><ul>
<li><b>status</b>: Resolved
<li><b>priority</b>: Critical
<li><b>component(s)</b>: matrix-project-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2017-04-19T08:40:05+00:00
<li><b>votes</b>: 25
<li><b>watchers</b>: 29
<li><b>imported</b>: 2025-12-06
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Reproduction case:</p>

<p>Create a concurrent matrix job with a user defined axis that does a 'sleep 120' as its build step. Launch several of these jobs, enough that all available executors are taken up and there are still builds in the queue. Some of these builds will abort, with a console message similar to:</p>

<p><span class="error">&#91;...&#93;</span>
9 completed with result SUCCESS
24 completed with result SUCCESS
23 completed with result SUCCESS
2 completed with result SUCCESS
10 appears to be cancelled
10 completed with result ABORTED
25 appears to be cancelled
25 completed with result ABORTED
18 appears to be cancelled
18 completed with result ABORTED
13 appears to be cancelled
<span class="error">&#91;...&#93;</span></p>

<p>For my test case, I have 26 slaves, 82 executors, 25 sub-jobs. I can reproduce reliably if I launch 5 or more top level jobs at once.</p>

</pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>jkoleszar:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=163391&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-163391">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=163391&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-163391">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I was able to reproduce this by hacking one of Jenkins' unit tests as well:</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>diff --git a/test/src/test/groovy/hudson/matrix/MatrixProjectCustomWorkspaceTest.groovy b/test/src/test/groovy/hudson/matrix/MatrixProjectCustomWorkspaceTest.groovy
index 1ddb195..8b6f324 100644
--- a/test/src/test/groovy/hudson/matrix/MatrixProjectCustomWorkspaceTest.groovy
+++ b/test/src/test/groovy/hudson/matrix/MatrixProjectCustomWorkspaceTest.groovy
@@ -116,7 +116,7 @@ class MatrixProjectCustomWorkspaceTest extends HudsonTestCase {
      */
     def configureCustomWorkspaceConcurrentBuild(MatrixProject p) {
         // needs sufficient parallel execution capability
-        jenkins.numExecutors = 10
+        jenkins.numExecutors = 4
         jenkins.updateComputerList()
 
         p.axes = new AxisList(new TextAxis("foo", "1", "2"))
@@ -140,8 +140,10 @@ class MatrixProjectCustomWorkspaceTest extends HudsonTestCase {
         // get one going
         Thread.sleep(1000)
         def f2 = p.scheduleBuild2(0)
+        Thread.sleep(1000)
+        def f3 = p.scheduleBuild2(0)
 
-        def bs = [f1, f2]*.get().each { assertBuildStatusSuccess(it) }
+        def bs = [f1, f2, f3]*.get().each { assertBuildStatusSuccess(it) }
         return bs
     }
 }

</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>I was able to reproduce this by hacking one of Jenkins' unit tests as well:</p>


```
diff --git a/test/src/test/groovy/hudson/matrix/MatrixProjectCustomWorkspaceTest.groovy b/test/src/test/groovy/hudson/matrix/MatrixProjectCustomWorkspaceTest.groovy
index 1ddb195..8b6f324 100644
--- a/test/src/test/groovy/hudson/matrix/MatrixProjectCustomWorkspaceTest.groovy
+++ b/test/src/test/groovy/hudson/matrix/MatrixProjectCustomWorkspaceTest.groovy
@@ -116,7 +116,7 @@ class MatrixProjectCustomWorkspaceTest extends HudsonTestCase {
      */
     def configureCustomWorkspaceConcurrentBuild(MatrixProject p) {
 // needs sufficient parallel execution capability
-jenkins.numExecutors = 10
+jenkins.numExecutors = 4
 jenkins.updateComputerList()
 
 p.axes = new AxisList(new TextAxis("foo", "1", "2"))
@@ -140,8 +140,10 @@ class MatrixProjectCustomWorkspaceTest extends HudsonTestCase {
 // get one going
 Thread.sleep(1000)
 def f2 = p.scheduleBuild2(0)
+Thread.sleep(1000)
+def f3 = p.scheduleBuild2(0)
 
-def bs = [f1, f2]*.get().each { assertBuildStatusSuccess(it) }
+def bs = [f1, f2, f3]*.get().each { assertBuildStatusSuccess(it) }
 return bs
     }
 }


```

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/zioschild.png" title="zioschild's avatar" /> zioschild:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=164034&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-164034">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=164034&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-164034">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This seems to be a side effect of the fix of issue 6747.
The problem only appears when starting the matrix jobs in concurrent mode. Starting the job in serial mode will not abort the axis-jobs
Next problem: the workaround given in issue 6747 is not working anymore. So there is no possibility to patch the job to run them concurrent anymore.</p>
</pre>
</details>

</ul>
</details>

<p>This seems to be a side effect of the fix of issue 6747.<br/>
The problem only appears when starting the matrix jobs in concurrent mode. Starting the job in serial mode will not abort the axis-jobs<br/>
Next problem: the workaround given in issue 6747 is not working anymore. So there is no possibility to patch the job to run them concurrent anymore.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/trbaker.png" title="trbaker's avatar" /> trbaker:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=164653&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-164653">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=164653&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-164653">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Arg, I just wanted to starting using concurrent matrix builds and ran into this.  I am heartened to see that the bug is open and hope we can see resolution soon!</p>
</pre>
</details>

</ul>
</details>

<p>Arg, I just wanted to starting using concurrent matrix builds and ran into this.  I am heartened to see that the bug is open and hope we can see resolution soon!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=165686&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-165686">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=165686&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-165686">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Kohsuke Kawaguchi
Path:
 changelog.html
 core/src/main/java/hudson/matrix/MatrixBuild.java
 core/src/main/java/hudson/matrix/MatrixConfiguration.java
<a href="http://jenkins-ci.org/commit/jenkins/9c7ef619cc96dc0111220412e841199de71d5b8d" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/9c7ef619cc96dc0111220412e841199de71d5b8d</a>
Log:
  <span class="error">&#91;FIXED JENKINS-13972&#93;</span></p>

<p>Fixed a problem in actually making concurrent builds work.</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/c2c31e2b933a...9c7ef619cc96" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/c2c31e2b933a...9c7ef619cc96</a></p>

</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Kohsuke Kawaguchi<br/>
Path:<br/>
 changelog.html<br/>
 core/src/main/java/hudson/matrix/MatrixBuild.java<br/>
 core/src/main/java/hudson/matrix/MatrixConfiguration.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/9c7ef619cc96dc0111220412e841199de71d5b8d" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/9c7ef619cc96dc0111220412e841199de71d5b8d</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-13972&#93;</span></p>

<p>Fixed a problem in actually making concurrent builds work.</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/c2c31e2b933a...9c7ef619cc96" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/c2c31e2b933a...9c7ef619cc96</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dogfood.png" title="dogfood's avatar" /> dogfood:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=165716&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-165716">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=165716&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-165716">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/yellow.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/1812/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #1812</a>
     <span class="error">&#91;FIXED JENKINS-13972&#93;</span> (Revision 9c7ef619cc96dc0111220412e841199de71d5b8d)</p>

<p>     Result = UNSTABLE
Kohsuke Kawaguchi : <a href="https://github.com/jenkinsci/jenkins/commit/9c7ef619cc96dc0111220412e841199de71d5b8d" class="external-link" target="_blank" rel="nofollow noopener">9c7ef619cc96dc0111220412e841199de71d5b8d</a>
Files : </p>
<ul>
	<li>changelog.html</li>
	<li>core/src/main/java/hudson/matrix/MatrixConfiguration.java</li>
	<li>core/src/main/java/hudson/matrix/MatrixBuild.java</li>
</ul>

</pre>
</details>

</ul>
</details>

<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/yellow.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/1812/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #1812</a><br/>
     <span class="error">&#91;FIXED JENKINS-13972&#93;</span> (Revision 9c7ef619cc96dc0111220412e841199de71d5b8d)</p>

<p>     Result = UNSTABLE<br/>
Kohsuke Kawaguchi : <a href="https://github.com/jenkinsci/jenkins/commit/9c7ef619cc96dc0111220412e841199de71d5b8d" class="external-link" target="_blank" rel="nofollow noopener">9c7ef619cc96dc0111220412e841199de71d5b8d</a><br/>
Files : </p>
<ul>
	<li>changelog.html</li>
	<li>core/src/main/java/hudson/matrix/MatrixConfiguration.java</li>
	<li>core/src/main/java/hudson/matrix/MatrixBuild.java</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/aleksas.png" title="aleksas's avatar" /> aleksas:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=171104&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-171104">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=171104&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-171104">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Matrix build started on Debian Debian-5.0.9 downstream projects executed on amd64 and i386 debian 5.0 slaves.</p>

<p>no change for svn://************ since the previous build
Triggering buildnode_x86.deb50
Triggering buildnode_x86_64.deb50
Configuration buildnode_x86.deb50 is still in the queue: Waiting for next available executor on build-lnx32-2.deb50
buildnode_x86.deb50 completed with result SUCCESS
 appears to be cancelled
buildnode_x86_64.deb50 completed with result ABORTED
Notifying upstream build ************* #426 of job completion
All downstream projects complete!
Minimum result threshold not met for join project
Notifying upstream projects of job completion
Notifying upstream of completion: ********** #426
Finished: ABORTED</p>


<p>buildnode_x86_64.deb50 task log shows:</p>

<p>Notifying upstream projects of job completion
Finished: SUCCESS</p>


<p>Jenkins master runs on Windows Server 2008 
Jenkins ver. 1.492</p>

<p>slave java.version 1.6.0_0
master java.version	1.7</p>
</pre>
</details>

</ul>
</details>

<p>Matrix build started on Debian Debian-5.0.9 downstream projects executed on amd64 and i386 debian 5.0 slaves.</p>

<p>no change for svn://************ since the previous build<br/>
Triggering buildnode_x86.deb50<br/>
Triggering buildnode_x86_64.deb50<br/>
Configuration buildnode_x86.deb50 is still in the queue: Waiting for next available executor on build-lnx32-2.deb50<br/>
buildnode_x86.deb50 completed with result SUCCESS<br/>
 appears to be cancelled<br/>
buildnode_x86_64.deb50 completed with result ABORTED<br/>
Notifying upstream build ************* #426 of job completion<br/>
All downstream projects complete!<br/>
Minimum result threshold not met for join project<br/>
Notifying upstream projects of job completion<br/>
Notifying upstream of completion: ********** #426<br/>
Finished: ABORTED</p>


<p>buildnode_x86_64.deb50 task log shows:</p>

<p>Notifying upstream projects of job completion<br/>
Finished: SUCCESS</p>


<p>Jenkins master runs on Windows Server 2008 <br/>
Jenkins ver. 1.492</p>

<p>slave java.version 1.6.0_0<br/>
master java.version	1.7</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/aleksas.png" title="aleksas's avatar" /> aleksas:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=172555&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-172555">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=172555&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-172555">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Issue was reopened maybe due to <a href="https://issue-redirect.jenkins.io/issue/15587" title="Builds disappear from jobs - hudson.util.IOException2: Invalid directory name - java.text.ParseException: Unparseable date: &quot;39&quot;" class="issue-link" data-issue-key="JENKINS-15587"><del>JENKINS-15587</del></a> causing similar symptoms "job appears to be cancelled" in matrix build.
Judging only from jenkins system logs: date parse exception (<a href="https://issue-redirect.jenkins.io/issue/15587" title="Builds disappear from jobs - hudson.util.IOException2: Invalid directory name - java.text.ParseException: Unparseable date: &quot;39&quot;" class="issue-link" data-issue-key="JENKINS-15587"><del>JENKINS-15587</del></a>) preceded build abortion notification. </p>
</pre>
</details>

</ul>
</details>

<p>Issue was reopened maybe due to <a href="https://issue-redirect.jenkins.io/issue/15587" title="Builds disappear from jobs - hudson.util.IOException2: Invalid directory name - java.text.ParseException: Unparseable date: "39"" class="issue-link" data-issue-key="JENKINS-15587"><del>JENKINS-15587</del></a> causing similar symptoms "job appears to be cancelled" in matrix build.<br/>
Judging only from jenkins system logs: date parse exception (<a href="https://issue-redirect.jenkins.io/issue/15587" title="Builds disappear from jobs - hudson.util.IOException2: Invalid directory name - java.text.ParseException: Unparseable date: "39"" class="issue-link" data-issue-key="JENKINS-15587"><del>JENKINS-15587</del></a>) preceded build abortion notification. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=175290&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-175290">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=175290&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-175290">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Aleksas
Path:
 core/src/main/java/hudson/model/Run.java
<a href="http://jenkins-ci.org/commit/jenkins/3d850711bb1a31f11c4309bd798200fbc5410764" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/3d850711bb1a31f11c4309bd798200fbc5410764</a>
Log:
  Update core/src/main/java/hudson/model/Run.java</p>

<p>Handling NTFS symlinks introduced via Util.resolveSymlink.
<a href="https://issue-redirect.jenkins.io/issue/15587" title="Builds disappear from jobs - hudson.util.IOException2: Invalid directory name - java.text.ParseException: Unparseable date: &quot;39&quot;" class="issue-link" data-issue-key="JENKINS-15587"><del>JENKINS-15587</del></a>
Also probably culprit for <a href="https://issue-redirect.jenkins.io/issue/13972" title="Concurrent matrix builds abort" class="issue-link" data-issue-key="JENKINS-13972"><del>JENKINS-13972</del></a></p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Aleksas<br/>
Path:<br/>
 core/src/main/java/hudson/model/Run.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/3d850711bb1a31f11c4309bd798200fbc5410764" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/3d850711bb1a31f11c4309bd798200fbc5410764</a><br/>
Log:<br/>
  Update core/src/main/java/hudson/model/Run.java</p>

<p>Handling NTFS symlinks introduced via Util.resolveSymlink.<br/>
<a href="https://issue-redirect.jenkins.io/issue/15587" title="Builds disappear from jobs - hudson.util.IOException2: Invalid directory name - java.text.ParseException: Unparseable date: "39"" class="issue-link" data-issue-key="JENKINS-15587"><del>JENKINS-15587</del></a><br/>
Also probably culprit for <a href="https://issue-redirect.jenkins.io/issue/13972" title="Concurrent matrix builds abort" class="issue-link" data-issue-key="JENKINS-13972"><del>JENKINS-13972</del></a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>vanharen:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=175456&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-175456">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=175456&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-175456">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We are seeing this behavior back as well, we aren't using NTFS symlinks so not sure that recent change will address the issue for us</p>
</pre>
</details>

</ul>
</details>

<p>We are seeing this behavior back as well, we aren't using NTFS symlinks so not sure that recent change will address the issue for us</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/sarahwoodall.png" title="sarahwoodall's avatar" /> sarahwoodall:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=178454&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-178454">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=178454&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-178454">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I am seeing this issue for the first time today after upgrading to Jenkins 1.509.1 from the previous LTS version. I have a matrix job which runs four different flavours of build on each of three platforms (Mac, Linux and Windows). My job configuration has not changed (and in fact no code has been checked in at all since the last good build &#8211; I just started this build manually today to test Jenkins after the upgrade).
Our master is on Windows, and there are two Windows executors on the same machine. There are two Mac executors on a Mac slave, and two Linux executors on a Linux slave. All of the builds in fact complete successfully, but the master reports that all four of the Windows builds "appear to be cancelled" and then that they "completed with result ABORTED". UPDATE: I have seen similar behaviour for other matrix jobs, including some that do not run on the slaves at all. I think it is a matrix job issue, not a master/slave issue.
Changing the job configuration to make the builds run serially rather than in parallel appears to work round the problem.</p>

<p>UPDATE: I believe this problem is Windows only. On my Windows installation, I had to configure all my matrix jobs to run serially, so as to work round this bug. I have now moved my Jenkins master to a Mac, and I have changed all my jobs again so that they do not run serially. So far, I have not seen the problem occur even once on the Mac. (On the Mac I have Jenkins 1.509.2 installed, but I don't think there is a fix for anything like this between 1.509.1 and 1.509.2, so it's more likely to be the change of platform that has caused the improvement.)</p>
</pre>
</details>

</ul>
</details>

<p>I am seeing this issue for the first time today after upgrading to Jenkins 1.509.1 from the previous LTS version. I have a matrix job which runs four different flavours of build on each of three platforms (Mac, Linux and Windows). My job configuration has not changed (and in fact no code has been checked in at all since the last good build &#8211; I just started this build manually today to test Jenkins after the upgrade).<br/>
Our master is on Windows, and there are two Windows executors on the same machine. There are two Mac executors on a Mac slave, and two Linux executors on a Linux slave. All of the builds in fact complete successfully, but the master reports that all four of the Windows builds "appear to be cancelled" and then that they "completed with result ABORTED". UPDATE: I have seen similar behaviour for other matrix jobs, including some that do not run on the slaves at all. I think it is a matrix job issue, not a master/slave issue.<br/>
Changing the job configuration to make the builds run serially rather than in parallel appears to work round the problem.</p>

<p>UPDATE: I believe this problem is Windows only. On my Windows installation, I had to configure all my matrix jobs to run serially, so as to work round this bug. I have now moved my Jenkins master to a Mac, and I have changed all my jobs again so that they do not run serially. So far, I have not seen the problem occur even once on the Mac. (On the Mac I have Jenkins 1.509.2 installed, but I don't think there is a fix for anything like this between 1.509.1 and 1.509.2, so it's more likely to be the change of platform that has caused the improvement.)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ilatypov.png" title="ilatypov's avatar" /> ilatypov:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=179612&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-179612">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=179612&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-179612">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I see a machine that aborts a job on its second slave.  Both slaves start via SSH, and the machine runs a Centrify SSH server.</p>

<p>Other 2 machines run a regular SSH server and do not exhibit aborts on their second slaves.</p>

<p>We have Jenkins 1.492.</p>
</pre>
</details>

</ul>
</details>

<p>I see a machine that aborts a job on its second slave.  Both slaves start via SSH, and the machine runs a Centrify SSH server.</p>

<p>Other 2 machines run a regular SSH server and do not exhibit aborts on their second slaves.</p>

<p>We have Jenkins 1.492.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ilatypov.png" title="ilatypov's avatar" /> ilatypov:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=181222&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-181222">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=181222&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-181222">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I figured a node configuration of my matrix job received a "job disabled" property.  Sub-projects disabled via "Configuration Slicing/Job Disabled Build Slicer (bool)" in <tt>/slicing/jobdisabledbool/</tt> will deny requests for new runs without pointing the reason. </p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">[USER@MASTER ~]$ diff -u /usr/local/jenkins/data/jobs/MATRIXPROJ/configurations/axis-MATRIX/HOSTNAME{X,Y}/config.xml
--- /usr/local/jenkins/data/jobs/MATRIXPROJ/configurations/axis-MATRIX/HOSTNAMEX/config.xml  2013-06-06 21:43:25.823244000 -0400
+++ /usr/local/jenkins/data/jobs/MATRIXPROJ/configurations/axis-MATRIX/HOSTNAMEY/config.xml  2013-06-07 16:22:46.529940000 -0400
@@ -7,7 +7,7 @@
   &lt;/properties&gt;
   &lt;scm class=<span class="code-quote">"hudson.scm.NullSCM"</span>/&gt;
   &lt;canRoam&gt;<span class="code-keyword">true</span>&lt;/canRoam&gt;
-  &lt;disabled&gt;<span class="code-keyword">true</span>&lt;/disabled&gt;
+  &lt;disabled&gt;<span class="code-keyword">false</span>&lt;/disabled&gt;
   &lt;blockBuildWhenDownstreamBuilding&gt;<span class="code-keyword">false</span>&lt;/blockBuildWhenDownstreamBuilding&gt;
   &lt;blockBuildWhenUpstreamBuilding&gt;<span class="code-keyword">false</span>&lt;/blockBuildWhenUpstreamBuilding&gt;
   &lt;triggers class=<span class="code-quote">"vector"</span>/&gt;
</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>I figured a node configuration of my matrix job received a "job disabled" property.  Sub-projects disabled via "Configuration Slicing/Job Disabled Build Slicer (bool)" in <tt>/slicing/jobdisabledbool/</tt> will deny requests for new runs without pointing the reason. </p>


<pre>
[USER@​MASTER ~]$ diff -u /usr/local/jenkins/data/jobs/MATRIXPROJ/configurations/axis-MATRIX/HOSTNAME{X,Y}/config.xml
--- /usr/local/jenkins/data/jobs/MATRIXPROJ/configurations/axis-MATRIX/HOSTNAMEX/config.xml  2013-06-06 21:43:25.823244000 -0400
+++ /usr/local/jenkins/data/jobs/MATRIXPROJ/configurations/axis-MATRIX/HOSTNAMEY/config.xml  2013-06-07 16:22:46.529940000 -0400
@@ -7,7 +7,7 @@
   </properties>
   <scm class=<span class="code-quote">"hudson.scm.NullSCM"</span>/>
   <canRoam><span class="code-keyword">true</span></canRoam>
-  <disabled><span class="code-keyword">true</span></disabled>
+  <disabled><span class="code-keyword">false</span></disabled>
   <blockBuildWhenDownstreamBuilding><span class="code-keyword">false</span></blockBuildWhenDownstreamBuilding>
   <blockBuildWhenUpstreamBuilding><span class="code-keyword">false</span></blockBuildWhenUpstreamBuilding>
   <triggers class=<span class="code-quote">"vector"</span>/>
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>shay_weiss:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=195703&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-195703">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=195703&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-195703">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi all, </p>

<p>I've been investigating aborts issues in Jenkins and I've found at least one bug with regards to this.
Here is my report on the subject:
<a href="https://docs.google.com/presentation/d/1ybtB-Bhkb4c3dhb5ZMArr4prtEZ-pjLqH9Vk7yhdZTg/" class="external-link" target="_blank" rel="nofollow noopener">https://docs.google.com/presentation/d/1ybtB-Bhkb4c3dhb5ZMArr4prtEZ-pjLqH9Vk7yhdZTg/</a></p>

<p>There is also another issue I'm dealing with and in the process of investigating.</p>

<p>Core developers - I'll be happy to make a contribution to the sources if you can give me pointers on how to modify my proposed fix so it will be 'commit worthy' <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>Hi all, </p>

<p>I've been investigating aborts issues in Jenkins and I've found at least one bug with regards to this.<br/>
Here is my report on the subject:<br/>
<a href="https://docs.google.com/presentation/d/1ybtB-Bhkb4c3dhb5ZMArr4prtEZ-pjLqH9Vk7yhdZTg/" class="external-link" target="_blank" rel="nofollow noopener">https://docs.google.com/presentation/d/1ybtB-Bhkb4c3dhb5ZMArr4prtEZ-pjLqH9Vk7yhdZTg/</a></p>

<p>There is also another issue I'm dealing with and in the process of investigating.</p>

<p>Core developers - I'll be happy to make a contribution to the sources if you can give me pointers on how to modify my proposed fix so it will be 'commit worthy' <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>tizkiko:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=239007&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-239007">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=239007&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-239007">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi</p>

<p>Is the solution suggested above by Shay Weiss reasonable? is it going to be pushed in the next versions?</p>

<p>thanks</p>
</pre>
</details>

</ul>
</details>

<p>Hi</p>

<p>Is the solution suggested above by Shay Weiss reasonable? is it going to be pushed in the next versions?</p>

<p>thanks</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>tizkiko:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=254272&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254272">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=254272&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254272">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I created a pull request with a fix, can someone please review?
<a href="https://github.com/jenkinsci/matrix-project-plugin/pull/28" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/matrix-project-plugin/pull/28</a></p>

<p>thanks,
Tidhar</p>
</pre>
</details>

</ul>
</details>

<p>I created a pull request with a fix, can someone please review?<br/>
<a href="https://github.com/jenkinsci/matrix-project-plugin/pull/28" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/matrix-project-plugin/pull/28</a></p>

<p>thanks,<br/>
Tidhar</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/olivergondza.png" title="olivergondza's avatar" /> olivergondza:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=296211&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296211">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=296211&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296211">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Putting this back to fixed as confirmed in PR it is no longer a problem. When someone spot a similar problem, please, file new issue.</p>
</pre>
</details>

</ul>
</details>

<p>Putting this back to fixed as confirmed in PR it is no longer a problem. When someone spot a similar problem, please, file new issue.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>pjdarton:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/13972?focusedId=363576&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-363576">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-13972?focusedId=363576&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-363576">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>FYI someone did spot this again and raised <a href="https://issue-redirect.jenkins.io/issue/46453" title="matrix job with dynamic axis aborting builds" class="issue-link" data-issue-key="JENKINS-46453">JENKINS-46453</a></p>

<p>(and then one of my colleagues found that bug report after encountering the same symptoms, hence my interest in it)</p>
</pre>
</details>

</ul>
</details>

<p>FYI someone did spot this again and raised <a href="https://issue-redirect.jenkins.io/issue/46453" title="matrix job with dynamic axis aborting builds" class="issue-link" data-issue-key="JENKINS-46453">JENKINS-46453</a></p>

<p>(and then one of my colleagues found that bug report after encountering the same symptoms, hence my interest in it)</p>
